scripts/screeps_rl_experiment_card.py

```
@@ -633,6 +677,33 @@ def validate_card_supply(card: JsonObject) -> None:
             raise CardValidationError("consumed Loop A card supply requires consumed_by_report_id")
 
 
+def validate_source_gate(card: JsonObject) -> None:
+    raw = first_present(card, ("source_gate", "sourceGate"))
+    if raw is None:
+        return
+    if not isinstance(raw, dict):
+        raise CardValidationError("source_gate must be a JSON object")
+    if raw.get("type") not in (None, "screeps-rl-dataset-evaluation-gate"):
+        raise CardValidationError("source_gate.type must be screeps-rl-dataset-evaluation-gate")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)